app/Http/Controllers/AdminDash.php

```diff
@@ -415,64 +314,48 @@ public function updateController(Request $request, $id) {
                     $user->save();
                 }
             }
-        } elseif (Auth::user()->isAbleTo('train')) {
-            $user->del = $request->input('del');
+        }
+        if (Auth::user()->isAbleTo('roster') || Auth::user()->isAbleTo('train')) { // Update training certifications
+            $user->clt_gnd = $request->input('clt_gnd');
+            $user->clt_twr = $request->input('clt_twr');
+            $user->clt_app = $request->input('clt_app');
+            $user->atl_gnd = $request->input('atl_gnd');
+            $user->atl_twr = $request->input('atl_twr');
+            $user->atl_app = $request->input('atl_app');
+
             $user->gnd = $request->input('gnd');
-            if ($user->twr == 99) {
-                if ($request->input('twr') != 0) {
-                    $solo = SoloCert::where('cid', $user->id)->where('status', 0)->first();
-                    if ($solo) {
-                        $solo->status = 1;
-                        $solo->save();
+            $positions = ['twr','app','ctr'];
+            foreach ($positions as $solo_id => $position) {
+                if ($user[$position] == 99) {
+                    if ($request->input($position) != 0) {
+                        $solo = SoloCert::where('cid', $user->id)->where('status', 0)->first();
+                        if ($solo) {
+                            $solo->status = 1;
+                            $solo->save();
+                        }
+                        $user[$position] = $request->input($position);
+                    } else {
+                        $user[$position] = 99;
                     }
-                    $user->twr = $request->input('twr');
+                } elseif ($request->input($position) == 99) {
+                    $user[$position] = $request->input($position);
+                    $expire = Carbon::now()->addMonth()->format('Y-m-d');
```

Comment:
We should do validation on this. Per GCAP 7.3(c), this can be no less than 7 days and no more than 90 days.

Reply:
It's automatic 1-month per our training department's policy

Reply:
30 days for all solo certs

Reply:
Updated